package project3authentication.auth.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;


@Schema(description = "New user registration details")
public record RegisterRequestDto(@NotBlank(message = "Name is required") String name,

                                 @NotBlank(message = "Email is required") @Email(message = "Invalid email format") String email,

                                 @NotBlank(message = "Password is required") @Size(min = 6, message = "Password must be at least 6 characters") String password) {
}
